processor/src/main/java/org/mapstruct/ap/internal/model/common/TypeFactoryUtil.java

```diff
@@ -0,0 +1,36 @@
+/**
+ *  Copyright 2012-2015 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.internal.model.common;
+
+final class TypeFactoryUtil {
+
+    private TypeFactoryUtil() {
+    }
+
+    static String trimSimpleClassName(String className) {
```

Comment:
Here's the thought flow:

1. Of course at first this method can be private method inside TypeFactory class.
2. However I want it unit tested, therefore I need to have access to it from tests, therefore I increase access from private to package protected.
3. At this point it feels better to have it as separate package protected utility class.

Of course we can stop at point 2 and still have testable code, going to point 3 is more preference, than necessity.